test/framework/utilities_test.py

```diff
@@ -191,6 +191,23 @@ def test_LooseVersion(self):
         self.assertEqual(LooseVersion('2.a').version, [2, 'a'])
         self.assertEqual(LooseVersion('2.a5').version, [2, 'a', 5])
 
+    def test_unique_ordered_append(self):
+        """Test unique_ordered_list_append method"""
+        base = ["potato", "tomato", "orange"]
+
+        reference = ["potato", "tomato", "orange", "apple"]
+        self.assertEqual(tu.unique_ordered_append(base, ["apple"]), reference)
```

Comment:
Is it intended, that the original list gets modified? I.e. after this line `base == ["potato", "tomato", "orange", "apple"]`

If this is intended I'd add a test case showing it and maybe rather pass in copies and assert the base is not modified:
```suggestion
        self.assertEqual(tu.unique_ordered_append(base.copy(), ["apple"]), reference)
        # Or
        base_orig = base.copy()
        self.assertEqual(tu.unique_ordered_append(base, ["apple"]), reference)
        self.assertEqual(base, base_orig)
```